jujugui/static/gui/src/app/components/deployment-flow/credential/credential.js

```diff
@@ -77,30 +75,57 @@ YUI.add('deployment-credential', function() {
       @method _getCredentials
     */
     _getCredentials: function() {
-      this.credentialXHR = this.props.listTemplates(
-        this._getCredentialsCallback);
+      const cloud = this.props.cloud;
+      const user = this.props.users.jem.user;
+      if (user) {
+        this.props.getTagsForCloudCredentials(
+          [[user, `cloud-${cloud}`]],
+          this._getTagsCallback);
+      }
     },
 
     /**
-      The method to be called when the credentials reponse has been received.
+      The method to be called when the credentials tags reponse has been
+      received.
+
+      @method _getTagsCallback
+      @param {String} error An error message, or null if there's no error.
+      @param {Array} tags A list of the tags found.
+    */
+    _getTagsCallback: function(error, tags) {
+      if (error) {
+        console.error('Unable to get tags for credentials', error);
+        return;
+      }
+      const tagList = [];
+      tags.forEach(tag => {
+        tagList = tagList.concat(tag.tags);
```

Comment:
From the docstring of `getTagsForCloudCredentials`: "in the sequence of results each result refers to the corresponding provided user/cloud pair". This is probably not clear enough, and that's my fault, but the meaning is that if you have a call like getTagsForCloudCredentials([pair1, pair2, pair3]) the callback will receive an array of results like [{tags: tags for pair1}, {tags: tags for pair2}, {tags: tags for pair3}], so each results refers to the user/cloud pair at that same index in the input tags. I guess this is less important now because we will be passing a single user/cloud pair, but the code should make that explicit and justify the flattening IMHO.